File: node/consultar-ie.mjs
// Consulta a Inscrição Estadual (IE) de um CNPJ pela API da CNPJAPI (https://cnpjapi.com.br).
// Recurso premium (plano com IE). Requer Node.js 18+ (fetch nativo). Modulo ES (.mjs).
// Uso: CNPJAPI_KEY=cnpj_sua_chave node consultar-ie.mjs 00776574000156 SP
//      (a UF e opcional; sem ela, faz a varredura nacional)

const cnpj = process.argv[2] ?? "00776574000156";
const uf = process.argv[3] ?? "";
const apiKey = process.env.CNPJAPI_KEY ?? "cnpj_sua_chave";

const url = new URL(`https://api.cnpjapi.com.br/consulta/ie/${cnpj}`);
if (uf) url.searchParams.set("uf", uf);

const resposta = await fetch(url, {
  headers: { Authorization: `Bearer ${apiKey}` },
});

if (!resposta.ok) {
  console.error(`Falha na consulta de IE: HTTP ${resposta.status}`);
  process.exit(1);
}

const { resultados } = await resposta.json();
if (resultados.length === 0) {
  console.log("Nenhuma inscricao estadual encontrada.");
} else {
  for (const ie of resultados) {
    console.log(`${ie.uf}: ${ie.ie || "(nao-contribuinte)"} - ${ie.situacao}`);
  }
}

File: python/consultar_ie.py
"""Consulta a Inscrição Estadual (IE) de um CNPJ pela API da CNPJAPI (https://cnpjapi.com.br).

Recurso premium (plano com IE). Requer: pip install requests
Uso:    CNPJAPI_KEY=cnpj_sua_chave python consultar_ie.py 00776574000156 SP
        (a UF e opcional; sem ela, faz a varredura nacional)
"""

import os
import sys

import requests


def consultar_ie(cnpj: str, api_key: str, uf: str = "") -> dict:
    params = {"uf": uf} if uf else {}
    resposta = requests.get(
        f"https://api.cnpjapi.com.br/consulta/ie/{cnpj}",
        params=params,
        headers={"Authorization": f"Bearer {api_key}"},
        timeout=15,
    )
    resposta.raise_for_status()
    return resposta.json()


def main() -> None:
    cnpj = sys.argv[1] if len(sys.argv) > 1 else "00776574000156"
    uf = sys.argv[2] if len(sys.argv) > 2 else ""
    api_key = os.environ.get("CNPJAPI_KEY", "cnpj_sua_chave")

    dados = consultar_ie(cnpj, api_key, uf)
    if not dados["resultados"]:
        print("Nenhuma inscricao estadual encontrada.")
        return
    for ie in dados["resultados"]:
        numero = ie["ie"] or "(nao-contribuinte)"
        print(f"{ie['uf']}: {numero} - {ie['situacao']}")


if __name__ == "__main__":
    main()
